src/v/redpanda/debug_bundle.cc

```diff
@@ -0,0 +1,397 @@
+    return bundle_name.string();
+}
+
+ss::future<std::error_code> debug_bundle::is_ready(ss::sstring& filename) {
+    // First check if the requested filename is actually a bundle
+    // TODO(@NyaliaLui): In the future when we support more than 1 bundle on
+    // disk, this condition will evolve to check if the filename follows our
+    // naming scheme.
+    if (filename != default_bundle_name()) {
+        co_return rpk::debug_bundle_error::invalid_filename;
+    }
+
+    // Then check if the requested filename is on disk
+    auto bundle_name = detail::make_bundle_filename(get_write_dir(), filename);
+    auto bundle_status = co_await get_status();
+    if (!std::filesystem::exists(bundle_name.c_str())) {
+        if (bundle_status == debug_bundle_status::running) {
+            co_return rpk::debug_bundle_error::bundle_not_ready;
+        } else {
+            co_return rpk::debug_bundle_error::bundle_not_on_disk;
+        }
+    } else {
+        // It is possible for the status check to occur in a small window where
+        // the spawned process is still running but RPK has written to disk
+        // already.
+        if (bundle_status == debug_bundle_status::running) {
+            co_return rpk::debug_bundle_error::bundle_not_ready;
+        }
+    }
+
+    co_return rpk::debug_bundle_error::success;
+}
+
+rpk::debug_bundle_error detail::make_rpk_err_value(std::error_code err) {
+    return rpk::debug_bundle_error(err.value());
+}
+
+ss::future<std::error_code> debug_bundle::do_remove(ss::sstring& filename) {
+    co_await ss::remove_file(
+      detail::make_bundle_filename(get_write_dir(), filename));
+    co_await ss::sync_directory(get_write_dir().c_str());
+
+    co_return rpk::debug_bundle_error::success;
+}
+
+ss::future<std::error_code> debug_bundle::remove_bundle(ss::sstring& filename) {
+    gate_guard guard{_rpk_gate};
+    auto units = co_await _rpk_mu.get_units();
+
+    auto err = co_await is_ready(filename);
+    if (detail::make_rpk_err_value(err) != rpk::debug_bundle_error::success) {
+        co_return err;
+    }
+
+    co_return co_await container().invoke_on(
+      debug_bundle_shard_id,
+      [&filename](debug_bundle& b) { return b.do_remove(filename); });
+}
+
+ss::future<std::error_code> debug_bundle::do_nonblocking_abort() {
+    gate_guard guard{_rpk_gate};
+
+    if (!_process.has_value()) {
+        co_return rpk::debug_bundle_error::nothing_running;
+    }
+
+    // First try graceful stop
+    _process->terminate();
+    auto wstatus = co_await _process->wait();
+    auto err = detail::handle_wait_status(
+      wstatus, ss::sstring{"Graceful abort"}, default_bundle_name());
+    if (detail::make_rpk_err_value(err) != rpk::debug_bundle_error::success) {
+        // Do forceful stop if graceful failed
+        vlog(
+          bundle_log.debug,
+          "Failed to gracefully abort {}. Running SIGKILL ...",
+          default_bundle_name(),
+          err);
+        _process->kill();
```

Comment:
It may make more sense to wait for a set period of time before attempting to kill.

somethig like 
```c++
_process->terminate();
// some way of waitig for process to complete with a timer
// timer elapsed but process is still running
_process->kill();
// wait again with some timeout, but `kill` should really do it
```

Reply:
LEt's rubberduck this because:
1. I think we should avoid puting waits/sleeps into our code as much as possible
2. IMO, we should only call sigkill if sigterm failed, but how can we detect a failed sigterm?

Reply:
>IMO, we should only call sigkill if sigterm failed, but how can we detect a failed sigterm?

you can't. signal delivery happens, then you can observe the process. if it exited, then great. but if it i didn't exit, how long should you wait until you send it a sigkill?